### DISCUSSION TITLE
Remove the `Promise` polyfill

### DISCUSSION THREAD

**Discussion by dead-claudia** [MithrilJS/mithril.js#2742] [Ideas]
**Mithril version:**

**Platform and OS:**

**Project:**

**Is this something you're interested in implementing yourself?** Yes

### Description

Remove the `Promise` polyfill

### Why

We're planning on dropping IE support soon, but even if we didn't drop it by v3, users could just as easily just ship a Promise polyfill of their own if they really needed it. It's outlived its use and isn't useful to most our consumers anymore.

### Possible Implementation

1. Remove the ES promise polyfill
2. Switch `m.request` to just use the global

### Open Questions

Do we want to continue to support this legacy functionality? (I strongly say we shouldn't, and I anticipate most of us would agree here.)

**Comment by gilbert** [comment]
Promise polyfilling is now a well-trodden path, no need for mithril to include one 👍

**Comment by charlag** [comment]
For all the good, please do it! People who need to support IE11 in this day and age have to think pretty hard about polyfilling anyway.
Same goes for all the other XHR-related stuff which we have to ship right now.

**Comment by StephanHoyer** [comment]
It's done now
